android/src/main/java/com/urbanairship/reactnative/ReactMessageView.java

```diff
@@ -0,0 +1,161 @@
+
+    private WebViewClient webViewClient = new UAWebViewClient() {
+        private Integer error = null;
+
+        @Override
+        public void onPageFinished(WebView view, String url) {
+            super.onPageFinished(view, url);
+
+            if (message == null) {
+                return;
+            }
+
+            if (error != null) {
+                notifyError(message.getMessageId(), ERROR_MESSAGE_LOAD_FAILED, false);
+            } else {
+                message.markRead();
+                notifyLoadFinished(message.getMessageId());
+            }
+        }
+
+        @Override
+        public void onReceivedError(WebView view, int errorCode, String description, @Nullable String failingUrl) {
+            super.onReceivedError(view, errorCode, description, failingUrl);
+
+            if (message != null && failingUrl != null && failingUrl.equals(message.getMessageBodyUrl())) {
+                error = errorCode;
+            }
+        }
+
+        @Override
+        public void onClose(@NonNull WebView webView) {
+            if (message != null) {
+                notifyClose(message.getMessageId());
+            }
+        }
+    };
+
+    public ReactMessageView(@NonNull Context context) {
+        super(context);
+        setWebViewClient(webViewClient);
+    }
+
+    public void loadMessage(final String messageId) {
+        if (fetchMessageRequest != null) {
+            fetchMessageRequest.cancel();
+        }
+        this.message = null;
+
+        new Handler(Looper.getMainLooper()).post(new Runnable() {
+            @Override
+            public void run() {
+                startLoading(messageId);
+            }
+        });
+    }
+
+    void startLoading(final String messageId) {
+        notifyLoadStarted(messageId);
+
+        this.message = UAirship.shared().getInbox().getMessage(messageId);
+        if (this.message == null) {
+            fetchMessageRequest = UAirship.shared().getInbox().fetchMessages(new RichPushInbox.FetchMessagesCallback() {
+                @Override
+                public void onFinished(boolean success) {
+                    RichPushMessage message = UAirship.shared().getInbox().getMessage(messageId);
+                    if (!success) {
+                        notifyError(messageId, ERROR_FAILED_TO_FETCH_MESSAGE, true);
+                        return;
+                    } else if (message == null || message.isExpired()) {
+                        notifyError(messageId, ERROR_MESSAGE_NOT_AVAILABLE, false);
+                        return;
+                    }
+
+                    loadRichPushMessage(message);
+                }
+            });
+        } else {
+            if (this.message.isExpired()) {
+                notifyError(messageId, ERROR_MESSAGE_NOT_AVAILABLE, false);
+                return;
+            }
+            loadRichPushMessage(this.message);
+        }
+    }
+
+    private void notifyError(String messageId, String error, boolean retryable) {
```

Comment:
On iOS, it looks like this method is called `onLoadError`. Maybe `notifyLoadError()` here?